docs/src/devel/dev-env-setup.rst

```diff
@@ -178,3 +178,216 @@ e.g.:
 ::
 
    OS_MIGRATE_FUNC_TEST_ARGS='--tags test_network,test_subnet --skip-tags test_clean_after' ./toolbox/run make test-func
+
+Running e2e tests
+-----------------
+
+os-migrate also has a suite of end to end, e2e, tests which tests a migration from one existing openstack deployment
+to another existing openstack deployment.
+
+You can also test with a single existing openstack deployment migrating from one project to another.
+
+These docs cover the testing of a single existing openstack deployment migrating from one project to another scenario.
+
+The concepts and prerequisites are the same for other deployments.
+
+Prerequisites for e2e test
+--------------------------
+
+* Source environment credentials
+* Destination environment credentials
+* Existing images in both source and destination environments
+* Flavors
+* Space requirements
+  - 2 images totalling 1.25GB
+  - 1 volume totalling 1GiB in source environment
+  - 2 volumes totalling 5GiB in destination environment
+  - 2 VMs totalling 35GB disk usage in each environment
+
+Below are the steps required to satisfy the above requirements and run e2e tests in a test environment, migrating
+resources from one project to another.
+
+Create source environment and destination environment projects and users
+------------------------------------------------------------------------
+
+.. code-block::
+
+    # Create the src user in the default domain with password 'redhat'
+    openstack user create --domain default --password redhat src
+
+    # Create the src project
+    openstack project create --domain default src
+
+    # Assign src user a 'member' role in the src project
+    openstack role add \
+    --user src --user-domain default \
+    --project src --project-domain default member
+
+    # Confirm role assignment was successful
+    openstack role assignment list --project src
+
+    # Create the dst user in the default domain with password 'redhat'
+    openstack user create --domain default --password redhat dst
+
+    # Create the dst project
+    openstack project create --domain default dst
+
+    # Assign dst user a 'member' role in the src project
+    openstack role add \
+    --user dst --user-domain default \
+    --project dst --project-domain default member
+
+    # Confirm role assignment was successful
+    openstack role assignment list --project dst
+
+Create images
+------------
+
+.. code-block::
+
+    # Download images
+    wget https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.2.2004-20200611.2.x86_64.qcow2
```

Comment:
Can we use stream instead? We are currently testing it in the CI but I already deploy it.

Reply:
Good idea. Now updated